src/components/MoneyRequestReportView/MoneyRequestReportTransactionList.tsx

```diff
@@ -82,6 +86,23 @@ function MoneyRequestReportTransactionList({report, transactions, reportActions,
 
     const {selectedTransactionsID, setSelectedTransactionsID, toggleTransaction, isTransactionSelected} = useMoneyRequestReportContext(report.reportID);
 
+    const focusedRoot = useRootNavigationState((state) => findFocusedRoute(state));
+
+    useEffect(() => {
+        if (focusedRoot?.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR || focusedRoot?.name === SCREENS.SEARCH.MONEY_REQUEST_REPORT) {
+            return;
+        }
+        setSelectedTransactionsID([]);
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
For each of the eslint disable we are trying to add a comment explaining why it has to be done - can you add it here? Can you also add it the eslint disable on one line only please